modules/operator/cloudinit.tf

```
@@ -101,6 +101,24 @@ data "cloudinit_config" "operator" {
     merge_type = local.default_cloud_init_merge_type
   }
 
+  # kubectl installation
+  dynamic "part" {
+    for_each = var.install_kubectl_from_repo ? [0] : [1]
```

```suggestion
    for_each = var.install_kubectl_from_repo ? [1] : []
```
Looks like the element in for_each clauses will have this run in both cases vs. empty else